tests/ignite/contrib/metrics/test_average_precision.py

```diff
@@ -318,7 +210,7 @@ def get_test_cases():
                 _test(y_pred, y, batch_size, idist.device())
 
 
-def _test_distrib_integration_binary(device):
+def _test_distrib_integration_binary_input(device):
```

Comment:
This test can be improved. The part binary_N and binary_N_L is mostly similar. The difference is the update function, so maybe write once in a test function taking the update function by argument. 

@KickItLikeShika What do you think ?

Reply:
Yes you're right! Thanks for pointing that out!